### ISSUE TITLE
_generate_random_poses of LLFF dataset is wrong

### ISSUE THREAD

**Issue by SJoJoK** [google-research/google-research#1352]
At the 1122 line of internal/datasets.py
```python
    # Generate random poses.
    random_poses = []
    cam2world = poses_avg(poses)
    up = poses[:, :3, 1].mean(0)
    for _ in range(n_poses):
      t = radii * np.concatenate([2 * np.random.rand(3) - 1., [1,]])
      position = cam2world @ t
      lookat = cam2world @ [0, 0, -focal, 1.]
      z_axis = position - lookat
    random_poses.append(viewmatrix(z_axis, up, position))
    self.random_poses = np.stack(random_poses, axis=0)
```
Its obvious that  `random_poses.append(viewmatrix(z_axis, up, position))` should in the for-loop, but it lies outside, which leads to only 1 random pose is genertated.

**Comment by amughrabi** [comment]
AFAICT, based on my testing, both cases have the same outcome.

**Comment by SJoJoK** [comment]
> AFAICT, based on my testing, both cases have the same outcome.

The metric may be the same. However, if `random_poses.append(viewmatrix(z_axis, up, position))` is in the for-loop. the `random_poses` will be a list contains `n_poses` of view-matrix, but in this case, it is a list which only contains one view-matrix.

**Comment by amughrabi** [comment]
True. A quick question. Can you replicate the results in the paper?

**Comment by SJoJoK** [comment]
Haven't tried yet, but I will do some experiments in the next few days. Is there any problem?

**Comment by amughrabi** [comment]
I can reproduce the results via the pre-trained models, but I cannot train custom data.

**Comment by amughrabi** [comment]
I can reproduce the results via the pre-trained models, but I cannot train custom data. Most likely, this is related to https://github.com/google-research/google-research/issues/1096

**Comment by SJoJoK** [comment]
> I can reproduce the results via the pre-trained models, but I cannot train custom data. Most likely, this is related to #1096

I'm sorry that I can't help you. I just ran this code a few times to understand some data processing flow. However, issues with custom datasets may not be related to #1096 since it just states that there is no NLL Loss in the code.

**Comment by amughrabi** [comment]
No problem, I figured it out. :)

**Comment by zhywanna** [comment]
so anyone knows how to calculate `RealNVP Loss`? @amughrabi @SJoJoK

**Comment by SJoJoK** [comment]
> so anyone knows how to calculate `RealNVP Loss`? @amughrabi @SJoJoK

sorry, idk.
